### PR TITLE
Improve the `uv sync` CLI documentation

### PR DESCRIPTION
**Pull request by zanieb**

### CODE REVIEW

File: crates/uv-cli/src/lib.rs [hunk 1]
```diff
@@ -567,6 +567,19 @@ pub enum ProjectCommand {
     )]
     Remove(RemoveArgs),
     /// Update the project's environment (experimental).
+    ///
+    /// Syncing ensures that all dependencies of the project are installed and
```
Comment (charliermarsh) [inline crates/uv-cli/src/lib.rs:571]: "project dependencies" instead of "dependencies of the project"

File: crates/uv-cli/src/lib.rs [hunk 2]
```diff
@@ -567,6 +567,19 @@ pub enum ProjectCommand {
...
+    /// up to date with the lockfile. Syncing also removes packages that are not
```
Comment (charliermarsh) [inline crates/uv-cli/src/lib.rs:572]: "up-to-date"

File: crates/uv-cli/src/lib.rs [hunk 3]
```diff
@@ -567,6 +567,19 @@ pub enum ProjectCommand {
...
+    /// declared as dependencies of the project.
+    ///
+    /// If the project virtual environment (`.venv`) does not exist, it will be
+    /// created.
+    ///
+    /// The project is re-locked before syncing unless the `--locked` or
+    /// `--frozen` flag is provided.
```
Comment (charliermarsh) [inline crates/uv-cli/src/lib.rs:579]: "flags are"?

File: crates/uv-cli/src/lib.rs [hunk 4]
```diff
@@ -2181,15 +2194,13 @@ pub struct RunArgs {
 #[derive(Args)]
 #[allow(clippy::struct_excessive_bools)]
 pub struct SyncArgs {
-    /// Include optional dependencies from the extra group name; may be provided more than once.
+    /// Include optional dependencies from the extra group name.
     ///
-    /// Only applies to `pyproject.toml`, `setup.py`, and `setup.cfg` sources.
+    /// May be provided more than once.
```
Comment (charliermarsh) [inline crates/uv-cli/src/lib.rs:2199]: I feel like this got removed from some other commands in previous PRs?

**Review by charliermarsh — APPROVED** [review]

File: crates/uv-cli/src/lib.rs (hunk 1 above)
Comment (zanieb) [reply crates/uv-cli/src/lib.rs:571]:
```suggestion
    /// Syncing ensures that all project dependencies are installed and
```

File: crates/uv-cli/src/lib.rs (hunk 2 above)
Comment (zanieb) [reply crates/uv-cli/src/lib.rs:572]:
```suggestion
    /// up-to-date with the lockfile. Syncing also removes packages that are not
```

File: crates/uv-cli/src/lib.rs (hunk 3 above)
Comment (zanieb) [reply crates/uv-cli/src/lib.rs:579]: Only one can be provided, idk e.g. it shortens to `unless the `--locked` flag is provided`. Could go either way, don't know what is "correct".

File: crates/uv-cli/src/lib.rs (hunk 4 above)
Comment (zanieb) [reply crates/uv-cli/src/lib.rs:2199]: I've usually moved it out of the first line but retained it in general. I can audit again at the end.

**Pull request merged**